File: MiniDOMOffice/Office/Dispositivi/DeviceLog/DeviceLogCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {


        /// <summary>
        /// Cursore sulla tabella dei log
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class DeviceLogCursor 
            : minidom.Databases.DBObjectCursorPO<DeviceLog>
        {
            private DBCursorField<int> m_IDDevice = new DBCursorField<int>("IDDevice");
            private DBCursorField<DeviceLogFlags> m_Flags = new DBCursorField<DeviceLogFlags>("Flags");
            private DBCursorField<DateTime> m_StartDate = new DBCursorField<DateTime>("StartDate");
            private DBCursorField<DateTime> m_EndDate = new DBCursorField<DateTime>("EndDate");
            private DBCursorField<int> m_IDUtente = new DBCursorField<int>("IDUtente");
            private DBCursorStringField m_NomeUtente = new DBCursorStringField("NomeUtente");
            private DBCursorField<int> m_CPUUsage = new DBCursorField<int>("CPUUsage");
            private DBCursorField<int> m_CPUMaximum = new DBCursorField<int>("CPUMaximum");
            private DBCursorField<double> m_RAMTotal = new DBCursorField<double>("RAMTotal");
            private DBCursorField<double> m_RAMAvailable = new DBCursorField<double>("RAMAvailable");
            private DBCursorField<double> m_RAMMinimum = new DBCursorField<double>("RAMMinimum");
            private DBCursorField<double> m_DiskTotal = new DBCursorField<double>("DiskTotal");
            private DBCursorField<double> m_DiskAvailable = new DBCursorField<double>("DiskAvailable");
            private DBCursorField<double> m_DiskMinimum = new DBCursorField<double>("DiskMinimum");
            private DBCursorField<float> m_Temperature = new DBCursorField<float>("Temperature");
            private DBCursorField<float> m_TemperatureMaximum = new DBCursorField<float>("TemperatureMaximum");
            private DBCursorField<int> m_Counter1 = new DBCursorField<int>("Counter1");
            private DBCursorField<int> m_Counter2 = new DBCursorField<int>("Counter2");
            private DBCursorField<int> m_Counter3 = new DBCursorField<int>("Counter3");
            private DBCursorField<int> m_Counter4 = new DBCursorField<int>("Counter4");
            private DBCursorField<int> m_NumeroCampioni = new DBCursorField<int>("NumeroCampioni");
            private DBCursorField<int> m_ScreenColors = new DBCursorField<int>("ScreenColors");
            private DBCursorField<double> m_GPS_Alt = new DBCursorField<double>("GPS_Alt");
            private DBCursorField<double> m_GPS_Lat = new DBCursorField<double>("GPS_Lat");
            private DBCursorField<double> m_GPS_Lon = new DBCursorField<double>("GPS_Lon");
            private DBCursorField<double> m_GPS_Bear = new DBCursorField<double>("GPS_Bear");
            private DBCursorField<int> m_Screen_Width = new DBCursorField<int>("Screen_Width");
            private DBCursorField<int> m_Screen_Height = new DBCursorField<int>("Screen_Height");
            private DBCursorStringField m_IPAddress = new DBCursorStringField("IPAddress");
            private DBCursorStringField m_MACAddress = new DBCursorStringField("MACAddress");
            private DBCursorStringField m_OSVersion = new DBCursorStringField("OSVersion");
            private DBCursorStringField m_DettaglioStato = new DBCursorStringField("DettaglioStato");

            /// <summary>
            /// Costruttore
            /// </summary>
            public DeviceLogCursor()
            {
            }

            /// <summary>
            /// IPAddress
            /// </summary>
            public DBCursorStringField IPAddress
            {
                get
                {
                    return m_IPAddress;
                }
            }

            /// <summary>
            /// MACAddress
            /// </summary>
            public DBCursorStringField MACAddress
            {
                get
                {
                    return m_MACAddress;
                }
            }

            /// <summary>
            /// OSVersion
            /// </summary>
            public DBCursorStringField OSVersion
            {
                get
                {
                    return m_OSVersion;
                }
            }

            /// <summary>
            /// DettaglioStato
            /// </summary>
            public DBCursorStringField DettaglioStato
            {
                get
                {
                    return m_DettaglioStato;
                }
            }

            /// <summary>
            /// IDDevice
            /// </summary>
            public DBCursorField<int> IDDevice
            {
                get
                {
                    return m_IDDevice;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<DeviceLogFlags> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// StartDate
            /// </summary>
            public DBCursorField<DateTime> StartDate
            {
                get
                {
                    return m_StartDate;
                }
            }

            /// <summary>
            /// EndDate
            /// </summary>
            public DBCursorField<DateTime> EndDate
            {
                get
                {
                    return m_EndDate;
                }
            }

            /// <summary>
            /// IDUtente
            /// </summary>
            public DBCursorField<int> IDUtente
            {
                get
                {
                    return m_IDUtente;
                }
            }

            /// <summary>
            /// NomeUtente
            /// </summary>
            public DBCursorStringField NomeUtente
            {
                get
                {
                    return m_NomeUtente;
                }
            }

            /// <summary>
            /// CPUUsage
            /// </summary>
            public DBCursorField<int> CPUUsage
            {
                get
                {
                    return m_CPUUsage;
                }
            }

            /// <summary>
            /// CPUMaximum
            /// </summary>
            public DBCursorField<int> CPUMaximum
            {
                get
                {
                    return m_CPUMaximum;
                }
            }

            /// <summary>
            /// RAMTotal
            /// </summary>
            public DBCursorField<double> RAMTotal
            {
                get
                {
                    return m_RAMTotal;
                }
            }

            /// <summary>
            /// RAMAvailable
            /// </summary>
            public DBCursorField<double> RAMAvailable
            {
                get
                {
                    return m_RAMAvailable;
                }
            }

            /// <summary>
            /// RAMMinimum
            /// </summary>
            public DBCursorField<double> RAMMinimum
            {
                get
                {
                    return m_RAMMinimum;
                }
            }

            /// <summary>
            /// DiskTotal
            /// </summary>
            public DBCursorField<double> DiskTotal
            {
                get
                {
                    return m_DiskTotal;
                }
            }

            /// <summary>
            /// DiskAvailable
            /// </summary>
            public DBCursorField<double> DiskAvailable
            {
                get
                {
                    return m_DiskAvailable;
                }
            }

            /// <summary>
            /// DiskMinimum
            /// </summary>
            public DBCursorField<double> DiskMinimum
            {
                get
                {
                    return m_DiskMinimum;
                }
            }

            /// <summary>
            /// Temperature
            /// </summary>
            public DBCursorField<float> Temperature
            {
                get
                {
                    return m_Temperature;
                }
            }

            /// <summary>
            /// TemperatureMaximum
            /// </summary>
            public DBCursorField<float> TemperatureMaximum
            {
                get
                {
                    return m_TemperatureMaximum;
                }
            }

            /// <summary>
            /// Counter1
            /// </summary>
            public DBCursorField<int> Counter1
            {
                get
                {
                    return m_Counter1;
                }
            }

            /// <summary>
            /// Counter2
            /// </summary>
            public DBCursorField<int> Counter2
            {
                get
                {
                    return m_Counter2;
                }
            }

            /// <summary>
            /// Counter3
            /// </summary>
            public DBCursorField<int> Counter3
            {
                get
                {
                    return m_Counter3;
                }
            }

            /// <summary>
            /// Counter4
            /// </summary>
            public DBCursorField<int> Counter4
            {
                get
                {
                    return m_Counter4;
                }
            }

            /// <summary>
            /// NumeroCampioni
            /// </summary>
            public DBCursorField<int> NumeroCampioni
            {
                get
                {
                    return m_NumeroCampioni;
                }
            }

            /// <summary>
            /// ScreenColors
            /// </summary>
            public DBCursorField<int> ScreenColors
            {
                get
                {
                    return m_ScreenColors;
                }
            }

            /// <summary>
            /// GPS_Alt
            /// </summary>
            public DBCursorField<double> GPS_Alt
            {
                get
                {
                    return m_GPS_Alt;
                }
            }

            /// <summary>
            /// GPS_Lat
            /// </summary>
            public DBCursorField<double> GPS_Lat
            {
                get
                {
                    return m_GPS_Lat;
                }
            }

            /// <summary>
            /// GPS_Lon
            /// </summary>
            public DBCursorField<double> GPS_Lon
            {
                get
                {
                    return m_GPS_Lon;
                }
            }

            /// <summary>
            /// GPS_Bear
            /// </summary>
            public DBCursorField<double> GPS_Bear
            {
                get
                {
                    return m_GPS_Bear;
                }
            }

            /// <summary>
            /// Screen_Width
            /// </summary>
            public DBCursorField<int> Screen_Width
            {
                get
                {
                    return m_Screen_Width;
                }
            }

            /// <summary>
            /// Screen_Height
            /// </summary>
            public DBCursorField<int> Screen_Height
            {
                get
                {
                    return m_Screen_Height;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.DevicesLog;
            }
             
        }
    }
}

File: MiniDOMOffice/Office/Dispositivi/DeviceLog/DeviceLogCursor.vb
Imports minidom.Databases
Imports minidom.Sistema

Partial Class Office


    ''' <summary>
    ''' Cursore sulla tabella dei log
    ''' </summary>
    ''' <remarks></remarks>
    Public Class DeviceLogCursor
        Inherits DBObjectCursorPO(Of DeviceLog)

        Private m_IDDevice As New CCursorField(Of Integer)("IDDevice")
        Private m_Flags As New CCursorField(Of DeviceLogFlags)("Flags")
        Private m_StartDate As New CCursorField(Of Date)("StartDate")
        Private m_EndDate As New CCursorField(Of Date)("EndDate")

        Private m_IDUtente As New CCursorField(Of Integer)("IDUtente")
        Private m_NomeUtente As New CCursorFieldObj(Of String)("NomeUtente")

        Private m_CPUUsage As New CCursorField(Of Integer)("CPUUsage")
        Private m_CPUMaximum As New CCursorField(Of Integer)("CPUMaximum")

        Private m_RAMTotal As New CCursorField(Of Double)("RAMTotal")
        Private m_RAMAvailable As New CCursorField(Of Double)("RAMAvailable")
        Private m_RAMMinimum As New CCursorField(Of Double)("RAMMinimum")

        Private m_DiskTotal As New CCursorField(Of Double)("DiskTotal")
        Private m_DiskAvailable As New CCursorField(Of Double)("DiskAvailable")
        Private m_DiskMinimum As New CCursorField(Of Double)("DiskMinimum")

        Private m_Temperature As New CCursorField(Of Single)("Temperature")
        Private m_TemperatureMaximum As New CCursorField(Of Single)("TemperatureMaximum")

        Private m_Counter1 As New CCursorField(Of Integer)("Counter1")
        Private m_Counter2 As New CCursorField(Of Integer)("Counter2")
        Private m_Counter3 As New CCursorField(Of Integer)("Counter3")
        Private m_Counter4 As New CCursorField(Of Integer)("Counter4")

        Private m_NumeroCampioni As New CCursorField(Of Integer)("NumeroCampioni")
        Private m_ScreenColors As New CCursorField(Of Integer)("ScreenColors")

        Private m_GPS_Alt As New CCursorField(Of Double)("GPS_Alt")
        Private m_GPS_Lat As New CCursorField(Of Double)("GPS_Lat")
        Private m_GPS_Lon As New CCursorField(Of Double)("GPS_Lon")
        Private m_GPS_Bear As New CCursorField(Of Double)("GPS_Bear")

        Private m_Screen_Width As New CCursorField(Of Integer)("Screen_Width")
        Private m_Screen_Height As New CCursorField(Of Integer)("Screen_Height")

        Private m_IPAddress As New CCursorFieldObj(Of String)("IPAddress")
        Private m_MACAddress As New CCursorFieldObj(Of String)("MACAddress")
        Private m_OSVersion As New CCursorFieldObj(Of String)("OSVersion")
        Private m_DettaglioStato As New CCursorFieldObj(Of String)("DettaglioStato")

        Public Sub New()
        End Sub

        Public ReadOnly Property IPAddress As CCursorFieldObj(Of String)
            Get
                Return Me.m_IPAddress
            End Get
        End Property

        Public ReadOnly Property MACAddress As CCursorFieldObj(Of String)
            Get
                Return Me.m_MACAddress
            End Get
        End Property

        Public ReadOnly Property OSVersion As CCursorFieldObj(Of String)
            Get
                Return Me.m_OSVersion
            End Get
        End Property

        Public ReadOnly Property DettaglioStato As CCursorFieldObj(Of String)
            Get
                Return Me.m_DettaglioStato
            End Get
        End Property

        Public ReadOnly Property IDDevice As CCursorField(Of Integer)
            Get
                Return Me.m_IDDevice
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of DeviceLogFlags)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property StartDate As CCursorField(Of Date)
            Get
                Return Me.m_StartDate
            End Get
        End Property

        Public ReadOnly Property EndDate As CCursorField(Of Date)
            Get
                Return Me.m_EndDate
            End Get
        End Property

        Public ReadOnly Property IDUtente As CCursorField(Of Integer)
            Get
                Return Me.m_IDUtente
            End Get
        End Property

        Public ReadOnly Property NomeUtente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeUtente
            End Get
        End Property

        Public ReadOnly Property CPUUsage As CCursorField(Of Integer)
            Get
                Return Me.m_CPUUsage
            End Get
        End Property

        Public ReadOnly Property CPUMaximum As CCursorField(Of Integer)
            Get
                Return Me.m_CPUMaximum
            End Get
        End Property

        Public ReadOnly Property RAMTotal As CCursorField(Of Double)
            Get
                Return Me.m_RAMTotal
            End Get
        End Property

        Public ReadOnly Property RAMAvailable As CCursorField(Of Double)
            Get
                Return Me.m_RAMAvailable
            End Get
        End Property

        Public ReadOnly Property RAMMinimum As CCursorField(Of Double)
            Get
                Return Me.m_RAMMinimum
            End Get
        End Property

        Public ReadOnly Property DiskTotal As CCursorField(Of Double)
            Get
                Return Me.m_DiskTotal
            End Get
        End Property

        Public ReadOnly Property DiskAvailable As CCursorField(Of Double)
            Get
                Return Me.m_DiskAvailable
            End Get
        End Property

        Public ReadOnly Property DiskMinimum As CCursorField(Of Double)
            Get
                Return Me.m_DiskMinimum
            End Get
        End Property

        Public ReadOnly Property Temperature As CCursorField(Of Single)
            Get
                Return Me.m_Temperature
            End Get
        End Property

        Public ReadOnly Property TemperatureMaximum As CCursorField(Of Single)
            Get
                Return Me.m_TemperatureMaximum
            End Get
        End Property

        Public ReadOnly Property Counter1 As CCursorField(Of Integer)
            Get
                Return Me.m_Counter1
            End Get
        End Property

        Public ReadOnly Property Counter2 As CCursorField(Of Integer)
            Get
                Return Me.m_Counter2
            End Get
        End Property

        Public ReadOnly Property Counter3 As CCursorField(Of Integer)
            Get
                Return Me.m_Counter3
            End Get
        End Property

        Public ReadOnly Property Counter4 As CCursorField(Of Integer)
            Get
                Return Me.m_Counter4
            End Get
        End Property

        Public ReadOnly Property NumeroCampioni As CCursorField(Of Integer)
            Get
                Return Me.m_NumeroCampioni
            End Get
        End Property

        Public ReadOnly Property ScreenColors As CCursorField(Of Integer)
            Get
                Return Me.m_ScreenColors
            End Get
        End Property

        Public ReadOnly Property GPS_Alt As CCursorField(Of Double)
            Get
                Return Me.m_GPS_Alt
            End Get
        End Property

        Public ReadOnly Property GPS_Lat As CCursorField(Of Double)
            Get
                Return Me.m_GPS_Lat
            End Get
        End Property

        Public ReadOnly Property GPS_Lon As CCursorField(Of Double)
            Get
                Return Me.m_GPS_Lon
            End Get
        End Property

        Public ReadOnly Property GPS_Bear As CCursorField(Of Double)
            Get
                Return Me.m_GPS_Bear
            End Get
        End Property

        Public ReadOnly Property Screen_Width As CCursorField(Of Integer)
            Get
                Return Me.m_Screen_Width
            End Get
        End Property

        Public ReadOnly Property Screen_Height As CCursorField(Of Integer)
            Get
                Return Me.m_Screen_Height
            End Get
        End Property


        Protected Overrides Function GetModule() As CModule
            Return Office.DevicesLog.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeDevLog"
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function

    End Class



End Class